dart-impl/mpi/src/dart_communication.c

```diff
@@ -667,7 +658,13 @@ dart_ret_t dart_put_handle(
                    "(from local allocation) "
                    "target_unit:%d offset:%"PRIu64"",
                    nelem, dtype, target_unitid_abs, offset);
```

Comment:
This is the "most portability-compliant" variant I know to print an `u64`.
Yes, it does look strange and ugly, I should have formatted this more nicely.
A space-separated series of strings is treated as a single string in C/C++:

~~~c
printf("Foo" "Bar" "Batz");
// same as
printf("FooBarBatz");
// so:
printf("offset: %"PRIu64"");
// is same as (assuming PRIu64 = "ul")
printf("offset: %ul");
~~~